src/NUnitFramework/framework/Internal/Results/TestSuiteResult.cs

```diff
@@ -220,74 +220,60 @@ public override IEnumerable<ITestResult> Children
         public virtual void AddResult(ITestResult result)
         {
 #if PARALLEL
-            var childrenAsConcurrentQueue = Children as ConcurrentQueue<ITestResult>;
-            if (childrenAsConcurrentQueue != null)
-                childrenAsConcurrentQueue.Enqueue(result);
```

Comment:
I had to modify this method because the way results were merged was not correct in the case of Theories. I refactored mostly so as to make sure I understood it. But the above lines are just wrong: if PARALLEL is defined, then Children is already a ConcurrentQueue - no need to convert it. If it's not defined, then it's a List. Somebody got mixed up at some time in the past, which is easy to do with conditional compilation.